src/main/java/com/microsoft/graph/content/BatchRequestContent.java

```diff
@@ -0,0 +1,282 @@
+                writeBatchRequestStepAsync(requestStep, writer);
+            }
+            writer.endArray();
+            writer.endObject();
+            writer.flush();
+            PipedInputStream in = new PipedInputStream();
+            try(final PipedOutputStream out = new PipedOutputStream(in)) {
+                outputStream.writeTo(out);
+                return CompletableFuture.completedFuture(in);
+            }
+        } catch(IOException e) {
+            CompletableFuture<InputStream> exception = new CompletableFuture<>();
+            exception.completeExceptionally(e);
+            return exception;
+        }
+    }
+    @SuppressFBWarnings
+    private void writeBatchRequestStepAsync(BatchRequestStep requestStep, JsonWriter writer) {
+        try {
+            Request request = requestStep.getRequest();
+            writer.beginObject();
+            writer.name(CoreConstants.BatchRequest.ID).value(requestStep.getRequestId());
+            writer.name(CoreConstants.BatchRequest.URL).value(getRelativeUrl(request.url()));
+            writer.name(CoreConstants.BatchRequest.METHOD).value(request.method());
+
+            List<String> dependsOn = requestStep.getDependsOn();
+            Headers headers = request.headers();
+            if(!dependsOn.isEmpty()) {
+                writer.name(CoreConstants.BatchRequest.DEPENDS_ON);
+                writer.beginArray();
+                for(String id : dependsOn) {
+                    writer.value(id);
+                }
+                writer.endArray();
+            }
+            RequestBody requestBody = request.body();
+            if(headers.size() != 0 || requestBody != null) {
+                writer.name(CoreConstants.BatchRequest.HEADERS);
+                writer.beginObject();
+                for(int i = 0; i < headers.size(); i++) {
+                    writer.name(headers.name(i)).value(headers.value(i));
+                }
+                if (requestBody != null) {
+                    writer.name("Content-Type").value(Objects.requireNonNull(requestBody.contentType()).toString());
+                    writer.endObject();
+                } else {
+                    writer.endObject();
+                }
+            }
+            if(requestBody != null) {
+                JsonObject bodyObject = getRequestContent(requestBody).join();
+                writer.name(CoreConstants.BatchRequest.BODY);
+                writeJsonElement(writer, bodyObject);
+            }
+            writer.endObject();
+        } catch (IOException e) {
+            CompletableFuture<Void> exception = new CompletableFuture<>();
+            exception.completeExceptionally(e);
+        }
+    }
+    private CompletableFuture<JsonObject> getRequestContent(RequestBody requestBody) {
+        try {
+            Buffer buffer = new Buffer();
+            requestBody.writeTo(buffer);
+            JsonObject jsonObject = JsonParser.parseString(buffer.readUtf8()).getAsJsonObject();
```

Comment:
This may be a mistake as the user may be sending a non-json payload as the request body(e.g uploading a file). We want the request content to be as it originally is. 
I will also need to fix this on the dotnet side 😅

Reply:
This method has been updated. Let me know what you think. Thanks!